Source/Core/Core/PowerPC/JitCommon/Jit_Util.cpp

```diff
@@ -510,24 +575,25 @@ void EmuCodeBlock::SafeWriteRegToReg(OpArg reg_value, X64Reg reg_addr, int acces
     }
   }
 
-  u32 mem_mask = Memory::ADDR_MASK_HW_ACCESS;
+  FixupBranch slow, exit;
+  if (!slowmem)
+  {
+    u32 mem_mask = Memory::ADDR_MASK_HW_ACCESS;
 
-  // The following masks the region used by the GC/Wii virtual memory lib
-  mem_mask |= Memory::ADDR_MASK_MEM1;
+    // The following masks the region used by the GC/Wii virtual memory lib
+    mem_mask |= Memory::ADDR_MASK_MEM1;
 
-  bool swap = !(flags & SAFE_LOADSTORE_NO_SWAP);
-
-  FixupBranch slow, exit;
-  slow = CheckIfSafeAddress(reg_value, reg_addr, registersInUse, mem_mask);
-  UnsafeWriteRegToReg(reg_value, reg_addr, accessSize, 0, swap);
-  if (farcode.Enabled())
-    SwitchToFarCode();
-  else
-    exit = J(true);
-  SetJumpTarget(slow);
+    slow = CheckIfSafeAddress(reg_value, reg_addr, registersInUse, mem_mask);
+    UnsafeWriteRegToReg(reg_value, reg_addr, accessSize, 0, swap);
+    if (farcode.Enabled())
+      SwitchToFarCode();
+    else
+      exit = J(true);
+    SetJumpTarget(slow);
 
-  // PC is used by memory watchpoints (if enabled) or to print accurate PC locations in debug logs
-  MOV(32, PPCSTATE(pc), Imm32(jit->js.compilerPC));
+    // PC is used by memory watchpoints (if enabled) or to print accurate PC locations in debug logs
+    MOV(32, PPCSTATE(pc), Imm32(jit->js.compilerPC));
```

Comment:
Good catch. We need to update PC when we end up on the slow path no matter what. I can fix this.